go-chaos/internal/network.go

```diff
@@ -15,150 +15,41 @@
 package internal
 
 import (
-	"bytes"
-	"context"
-	"errors"
-	"fmt"
 	"strings"
-
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/types"
 )
 
-func (c K8Client) ApplyNetworkPatch() error {
-
-	statefulSet, err := c.GetZeebeStatefulSet()
-	if err != nil {
-		return err
-	}
-
-	// We need to run the container with root to allow install tooling and give it network admin capabilities
-	patch := []byte(`{
-		"spec":{
-			"template":{
-				"spec":{
-					"containers":[
-						{
-							"name": "zeebe",
-							"securityContext":{
-								"runAsUser": 0,
-								"capabilities":{
-									"add":["NET_ADMIN"]
-								}
-							}
-						}]
-				}
-			}
-		}
-	}`)
-
-	_, err = c.Clientset.AppsV1().StatefulSets(c.GetCurrentNamespace()).Patch(context.TODO(), statefulSet.Name, types.StrategicMergePatchType, patch, metav1.PatchOptions{})
-	return err
-}
-
-func (c K8Client) ApplyNetworkPatchOnGateway() error {
-
-	deployment, err := c.getGatewayDeployment()
-	if err != nil {
-		return err
-	}
-
-	// We need to run the container with root to allow install tooling and give it network admin capabilities
-	patch := []byte(`{
-		"spec":{
-			"template":{
-				"spec":{
-					"containers":[
-						{
-							"name": "zeebe-gateway",
-							"securityContext":{
-								"runAsUser": 0,
-								"capabilities":{
-									"add":["NET_ADMIN"]
-								}
-							}
-						}]
-				}
-			}
-		}
-	}`)
-
-	_, err = c.Clientset.AppsV1().Deployments(c.GetCurrentNamespace()).Patch(context.TODO(), deployment.Name, types.StrategicMergePatchType, patch, metav1.PatchOptions{})
-	return err
-}
-
 func MakeIpUnreachableForPod(k8Client K8Client, podIp string, podName string) error {
-	// We try to reduce the system output in order to not break the execution. There is a limit for the sout for exec,
-	// for more details see remotecommand.StreamOptions
-
-	// the -qq flag makes the tool less noisy, remove it to get more output
-	err := k8Client.ExecuteCmdOnPod([]string{"apt", "-qq", "update"}, podName)
-	if err != nil {
-		return err
-	}
-
-	// the -qq flag makes the tool less noisy, remove it to get more output
-	err = k8Client.ExecuteCmdOnPod([]string{"apt", "-qq", "install", "-y", "iproute2"}, podName)
-	if err != nil {
-		return err
-	}
-
-	// we use replace to not break the execution, since add will return an exit code > 0 if the route exist
-	err = k8Client.ExecuteCmdOnPod([]string{"ip", "route", "replace", "unreachable", podIp}, podName)
-	if err != nil {
-		return err
-	}
-
-	return nil
+	cmd := []string{"ip", "route", "replace", "unreachable", podIp}
+	cmdWithSetup := []string{"sh", "-c", "apt update && apt install -y iproute2 && " + strings.Join(cmd, " ")}
+	var containerName string
+	if strings.Contains(podName, "gateway") {
```

Comment:
Is this to work with SaaS/SM?

Reply:
We have also functions to get the right pod name.

Reply:
This is to attach to the correct container, both on SaaS and self-managed. We already have the right `podName` here. Do you mean we have helper functions to get the right _container_ name?

Reply:
The pod name you use is actually only correct in SaaS in SM it is different but not sure whether you use the pod name here anyhow.

Reply:
This code here uses the pod name to figure out the correct _container_ name. If the pod name includes gateway, we assume that the container name is `zeebe-gateway`, otherwise we expect it to be `zeebe`. I think this should work in both SaaS and SM, right?